### ISSUE TITLE
TabView's New Tab Button color contrast ratio inaccessible (Light Theme)

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/microsoft-ui-xaml#2353]
**Describe the bug**
The New Tab Button for the TabView does not meet a 3:1 color contrast ratio.

**Steps to reproduce the bug**
This can be seen in Windows Terminal's [TabRowControl.xaml](https://github.com/microsoft/terminal/blob/master/src/cascadia/TerminalApp/TabRowControl.xaml), where we replace the NewTabButton with a SplitButton matching the styling of the TabView's NewTabButton.

More easily, this can be seen/tested in the Xaml Controls Gallery page introducing the TabView.

1. Open Xaml Controls Gallery and navigate to the TabView page
2. Open Accessibility Insights ([download link here](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-shcha1%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458847087032&sdata=uWx6cTEfazn%2BoCV3cL1iYB4KN%2BuXQ/MTlmOVec%2BZgKc%3D&reserved=0)) to check the color contrast ratio
3. Verify whether the color contrast ratio for the NewTabButton meets the 3:1 contrast ratio

**Expected behavior**
It meets the 3:1 ratio

**Screenshots**

- Light Theme (fails)
![image](https://user-images.githubusercontent.com/11050425/80767884-d3068400-8afd-11ea-842f-1e024e17e360.png)
![image](https://user-images.githubusercontent.com/11050425/80767928-e9acdb00-8afd-11ea-80fb-8d1d08ec8219.png)

- Dark Theme (passes)
![image](https://user-images.githubusercontent.com/11050425/80767901-d994fb80-8afd-11ea-8504-5a1e88605ba9.png)
![image](https://user-images.githubusercontent.com/11050425/80767945-efa2bc00-8afd-11ea-8099-82d27d2dab57.png)

**Version Info**

NuGet package version: 

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| November 2019 Update (18363)  |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 |    |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Comment by ranjeshj** [comment]
@carlos-zamora Both TabView + and the splitbutton have this issue ? @chigy as FYI.

**Comment by carlos-zamora** [comment]
> @carlos-zamora Carlos Zamora FTE Both TabView + and the splitbutton have this issue ? @chigy Chigusa Sansen FTE as FYI.

Yeah. I assume this is different from https://github.com/microsoft/microsoft-ui-xaml/issues/1901 because it's a different resource dictionary, but if not, feel free to close it.

This is being tracked in the Terminal repo here (https://github.com/microsoft/terminal/issues/5392) and was submitted by the accessibility testers.

**Comment by chigy** [comment]
@carlos-zamora , do you have screenshot of the offending UI?

**Comment by carlos-zamora** [comment]
> @carlos-zamora Carlos Zamora FTE , do you have screenshot of the offending UI?

Yes. They should be visible in the issue description above. Windows Terminal is on the left side, and Accessibility Insights is on the right detecting the issue.

**Comment by chigy** [comment]
@carlos-zamora , oops, I see it now. I wasn't clear what I was looking at. :)

@stmoy , is this issue with TabView or the way those resources are used within Terminal?

**Comment by carlos-zamora** [comment]
Huh. I just noticed that we do this:
```
                <mux:SplitButton.Resources>
		    <!-- Override the SplitButton* resources to match the tab view's button's styles. -->
                    <ResourceDictionary>
                        <ResourceDictionary.ThemeDictionaries>
                            <ResourceDictionary x:Key="Light">
                                <StaticResource x:Key="SplitButtonBackground" ResourceKey="TabViewItemHeaderBackground" />
                                <StaticResource x:Key="SplitButtonForeground" ResourceKey="SystemControlBackgroundBaseMediumBrush" />
                                <StaticResource x:Key="SplitButtonBackgroundPressed" ResourceKey="TabViewItemHeaderBackgroundPressed" />
                                <StaticResource x:Key="SplitButtonForegroundPressed" ResourceKey="SystemControlBackgroundBaseMediumHighBrush" />
                                <StaticResource x:Key="SplitButtonBackgroundPointerOver" ResourceKey="TabViewItemHeaderBackgroundPointerOver" />
                                <StaticResource x:Key="SplitButtonForegroundPointerOver" ResourceKey="SystemControlBackgroundBaseMediumHighBrush" />
                            </ResourceDictionary>
                            <ResourceDictionary x:Key="Dark">
                                <StaticResource x:Key="SplitButtonBackground" ResourceKey="TabViewItemHeaderBackground" />
                                <StaticResource x:Key="SplitButtonForeground" ResourceKey="SystemControlBackgroundBaseMediumBrush" />
                                <StaticResource x:Key="SplitButtonBackgroundPressed" ResourceKey="TabViewItemHeaderBackgroundPressed" />
                                <StaticResource x:Key="SplitButtonForegroundPressed" ResourceKey="SystemControlBackgroundBaseMediumHighBrush" />
                                <StaticResource x:Key="SplitButtonBackgroundPointerOver" ResourceKey="TabViewItemHeaderBackgroundPointerOver" />
                                <StaticResource x:Key="SplitButtonForegroundPointerOver" ResourceKey="SystemControlBackgroundBaseMediumHighBrush" />
                            </ResourceDictionary>
                            <ResourceDictionary x:Key="HighContrast">
                                <StaticResource x:Key="SplitButtonBackground" ResourceKey="TabViewItemHeaderBackground" />
                                <StaticResource x:Key="SplitButtonForeground" ResourceKey="SystemControlBackgroundBaseMediumBrush" />
                                <StaticResource x:Key="SplitButtonBackgroundPressed" ResourceKey="TabViewItemHeaderBackgroundPressed" />
                                <StaticResource x:Key="SplitButtonForegroundPressed" ResourceKey="SystemControlBackgroundBaseMediumHighBrush" />
                                <StaticResource x:Key="SplitButtonBackgroundPointerOver" ResourceKey="TabViewItemHeaderBackgroundPointerOver" />
                                <StaticResource x:Key="SplitButtonForegroundPointerOver" ResourceKey="SystemControlBackgroundBaseMediumHighBrush" />
                            </ResourceDictionary>
                        </ResourceDictionary.ThemeDictionaries>
                    </ResourceDictionary>
                </mux:SplitButton.Resources>
            </mux:SplitButton>
```
Overriding the SplitButton's resources to match the tab view's button styles.

So this might actually be an issue with the TabView resources. And a duplicate of  https://github.com/microsoft/microsoft-ui-xaml/issues/1901. Sorry!

**Comment by stmoy** [comment]
Good question. I was under the impression that the SplitButton was custom for Terminal (as the regular ol' TabView doesn't have one built in) - but I think that it cannot be fixed independently of the TabView fix since it has a transparent background. I don't think it's actually a dupe of #1901 as I'm guessing there will need to be some custom work here, but I think the former must be solved before this.

**Comment by carlos-zamora** [comment]
Ok, I went to the Xaml Controls Gallery and here are some more useful results:

Light Theme: fail
![light theme: fail](https://user-images.githubusercontent.com/11050425/80765094-04c81c80-8af7-11ea-8913-ff97873bc0bc.png)


Dark Theme: pass
![dark theme: pass](https://user-images.githubusercontent.com/11050425/80765120-16a9bf80-8af7-11ea-8e99-8f3a73bee3d9.png)

Note: the target ratio is 3:1 for UI components.

**Comment by chigy** [comment]
@carlos-zamora , what is this measuring?

**Comment by carlos-zamora** [comment]
Oh sorry! I've been running around a bit today. The above ratios are measuring the color contrast of the New Tab Button that is included by default in a TabView. Color 1 is the color of the '+' (foreground) , and Color 2 is the background.

These buttons are shown here:
![dark theme: new tab button](https://user-images.githubusercontent.com/11050425/80767368-938b6800-8afc-11ea-8142-5be3b9aee66c.png)![light theme: new tab button](https://user-images.githubusercontent.com/11050425/80767394-9f772a00-8afc-11ea-965d-56a91bf38533.png)

In Terminal, we suppress the New Tab button and replace it with a SplitButton that matches the styling (same foreground and background color).

So, it turns out, I was incorrect in what I first said reporting this issue. The problem isn't the SplitButton's color contrast, but rather the New Tab button's that is shipped in the TabView. I'll update the issue title and description appropriately. Sorry about the confusion.

**Comment by ranjeshj** [comment]
@stmoy @chigy  do we need to update the colors for the add/close buttons ?

**Comment by stmoy** [comment]
@ranjeshj - yes, but it's more than that. Several parts of TabView (background/foreground, buttons, text) don't meet the updated contrast ratio guidance. #1901 tracks taking a holistic pass across TabView to bring it up to compliance. We should probably close this one as a dupe of #1901 if @carlos-zamora believes that updating the defaults will fix this case.

**Comment by carlos-zamora** [comment]
Yeah. Feel free to mark this as a duplicate of that one :)

**Comment by ranjeshj** [comment]
ok. Lets track this as part of the other issue then. Thanks
